cmd/senpai/main.go

```diff
@@ -51,16 +53,23 @@ func main() {
 	lastNetID, lastBuffer := getLastBuffer()
 	app.SwitchToBuffer(lastNetID, lastBuffer)
 
+	// Write last buffer on close
+	sigCh := make(chan os.Signal, 1)
+	signal.Notify(sigCh, syscall.SIGINT, syscall.SIGTERM, syscall.SIGHUP)
+
+	go func() {
+		for sig := range sigCh {
```

Comment:
I believe this is an infinite loop. The `for` loop will continue to run after `SIGINT`/`SIGTERM` is handled. One way to fix this is to `return` after `app.Close()`, but since a `for` loop isn't really necessary in the first place, all of this function can be simplified to something among these lines:

```go
<-sigCh
app.Close()
```